dblookupext/miniblocksHandler.go

```diff
@@ -280,3 +263,112 @@ func (mh *miniblocksHandler) removeMiniblockMetadata(
 
 	return mh.saveMiniblocksMetadata(miniblockHash, loadedMiniblockMetadata, header.GetEpoch())
 }
+
+func (mh *miniblocksHandler) loadAllExistingMiniblocksMetadata(miniblockHash []byte, epoch uint32) (*MiniblockMetadataV2, error) {
+	miniblockMetadataFirstEpoch, err := mh.loadExistingMiniblocksMetadata(miniblockHash, epoch)
+	if err != nil {
+		return nil, err
+	}
+	numTxHashes := 0
+	for _, mbInfo := range miniblockMetadataFirstEpoch.MiniblocksInfo {
+		numTxHashes += len(mbInfo.TxHashesWhenPartial)
+	}
+	if numTxHashes == 0 || numTxHashes == int(miniblockMetadataFirstEpoch.NumTxHashesInMiniblock) {
+		return miniblockMetadataFirstEpoch, nil
+	}
+
+	// we will try the next epoch, maybe we will find the rest of the miniblockMetadata info
+	miniblockMetadataNextEpoch, err := mh.loadExistingMiniblocksMetadata(miniblockHash, epoch+1)
```

Comment:
can we have the data in epoch-1 ?

Reply:
no. The miniblock is saved only in the epoch where it was first started to be executed